optuna_dashboard/ts/components/TrialTable.tsx

```diff
@@ -0,0 +1,122 @@
+import ClearIcon from "@mui/icons-material/Clear"
+import DownloadIcon from "@mui/icons-material/Download"
+import FilterListIcon from "@mui/icons-material/FilterList"
+import {
+  Box,
+  Button,
+  Card,
+  CardContent,
+  IconButton,
+  InputAdornment,
+  Stack,
+  TextField,
+  useTheme,
+} from "@mui/material"
+import { TrialTable as TsLibTrialTable } from "@optuna/react"
+import { useAtomValue } from "jotai"
+import React, { FC, useState, useCallback } from "react"
+import { Link } from "react-router-dom"
+import { useConstants } from "../constantsProvider"
+import { useTrialFilterQuery } from "../hooks/useTrialFilterQuery"
+import { llmIsAvailable } from "../state"
+import { StudyDetail, Trial } from "../types/optuna"
+
+export const TrialTable: FC<{ studyDetail: StudyDetail }> = ({
+  studyDetail,
+}) => {
+  const { url_prefix } = useConstants()
+  const [trialFilter, render] = useTrialFilterQuery(5)
+  const [filterQuery, setFilterQuery] = useState("")
```

Comment:
We might use `useDeferredValue` for the debouncing here as mentioned at https://github.com/optuna/optuna-dashboard/pull/1127#discussion_r2244909636.

```suggestion
  const [_filterQuery, setFilterQuery] = useState("")
  const filterQuery = useDeferredValue(_filterQuery)
```